doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
+    >>> x = np.linspace(0, np.pi/2, 70)
+    >>> y = 1 / np.sqrt(1 - 0.5*np.sin(x)**2)
+    >>> spl = PchipInterpolator(x, y)
+
+and integrate
+
+    >>> spl.integrate(0, np.pi/2)
+    1.854074674965991
+
+which is indeed close to the value computed by `scipy.special.ellipk`.
+
+All piecewise polynomials can be constructed with N-dimensional ``y`` values.
+If ``y.ndim > 1``, it is understood as a stack of 1D ``y`` values, which are
+arranged along the interpolation axis (with the default value if 0).
+The latter is specified via the ``axis`` argument, and the invariant is that
+``len(x) == y.shape[axis]``. As an example, we extend the elliptic integral
+example above to compute the approximation for a range of ``m`` values, using
+the NumPy broadcasting:
+
+.. plot::
+
+    >>> from scipy.interpolate import PchipInterpolator
+    >>> m = np.linspace(0, 0.9, 11)
+    >>> x = np.linspace(0, np.pi/2, 70)
+    >>> y = 1 / np.sqrt(1 - m[:, None]*np.sin(x)**2)
+
+    Now the ``y`` array has the shape ``(11, 70)``, so that the values of ``y``
+    for fixed value of ``m`` are along the second axis of the ``y`` array.
+
+    >>> spl = PchipInterpolator(x, y, axis=1)  # the default is axis=0
+    >>> import matplotlib.pyplot as plt
+    >>> plt.plot(m, spl.integrate(0, np.pi/2), '--')
+
+    >>> from scipy.special import ellipk
+    >>> plt.plot(m, ellipk(m), 'o')
+    >>> plt.show()
+
+
+Interpolation with B-splines
+----------------------------
+
+A polynomial of degree :math:`k` can be thought of as a linear combination of
+:math:`k+1` monomial basis elements, :math:`1, x, x^2, \cdots, x^k`. B-splines
+form an alternative (if equivalent) :ref:`basis <tutorial-interpolate_bspl_basis>`
+of degree-:math:`k` piecewise polynomials. 
+
+As an illustration, we construct
+the interpolation of a :ref:`derivative of a sine function<tutorial-interpolate_ppoly>`
+with b-splines:
+
+.. plot::
+
+    >>> x = np.linspace(0, 3/2, 7)
+    >>> y = np.sin(np.pi*x)
+
+    To construct the interpolating objects given data arrays, ``x`` and ``y``,
+    we use the `make_interp_spline` function:
+
+    >>> from scipy.interpolate import make_interp_spline
+    >>> bspl = make_interp_spline(x, y)
+
+    Now ``bspl`` is a `BSpline` object which has an interface similar to `PPoly`.
+    In particular, it can be evaluated at a data point and differentiated:
+
+    >>> der = bspl.derivative()      # a BSpline representing the derivative
+    >>> import matplotlib.pyplot as plt
+    >>> xx = np.linspace(0, 3/2, 51)
+    >>> plt.plot(xx, bspl(xx), '--', label=r'$\sin(\pi x)$ approx')
+    >>> plt.plot(x, y, 'o', label='data')
+    >>> plt.plot(xx, der(xx)/np.pi, '--', label='$d \sin(\pi x)/dx / \pi$ approx')
+    >>> plt.legend()
+    >>> plt.show()
+
+Note that by default `make_interp_spline` constructs a cubic spline;
+its derivative is a quadratic:
+
+    >>> bspl.k, der.k
+    (3, 2)
+
+This way, the default result of ``make_interp_spline(x, y)`` is equivalent to
```

Comment:
Meant to mean as a continuation from the previous paragraph: "Note that by default `make_interp_spline` constructs a cubic spline... this way, the default result of ... is equivalent to ... ", but that obviously did not work. Tried changing it, would appreciate a read-over or a copy-edit to read better.

Reply:
Something I'm understanding now that I didn't understand before is that `CubicSpline` returns a `PPoly`, and `make_interp_spline` returns a `BSpline`.

Maybe you can help me fill in other blanks and I can suggest what might have helped me understand the first time.

It seems that `PPoly` and `BSpline` are the two fundamental spline representations that you want to cover in the tutorial, right?

Is there a function such that <function> is to `PPoly` as `make_interp_spline` is to `BSpline`? `CubicSpline` doesn't fulfill this relationship because it can't be of arbitrary order.

Reply:
The "this way" thing now reads "By default, the result of `make_interp_spline...", does it read better?

The difference between PPoly and BSpline : I tried to expand the section now that it's separate.